northstar/src/runtime/minijail.rs

```diff
@@ -214,27 +181,54 @@ impl Minijail {
             .collect::<Vec<String>>();
         let env = env.iter().map(|a| a.as_str()).collect::<Vec<&str>>();
 
+        let mut fds = vec![(self.log_fd, self.log_fd)];
+        let stdout = if let Some(io) = manifest.io.as_ref().and_then(|io| io.stdout.as_ref()) {
+            if *io == IoOutput::Pipe {
+                fds.push((1, 1));
+                None
+            } else {
+                let io = Io::new(io).await?;
+                fds.push((io.0, 1));
+                Some(io)
+            }
+        } else {
+            None
+        };
+        let stderr = if let Some(io) = manifest.io.as_ref().and_then(|io| io.stderr.as_ref()) {
+            if *io == IoOutput::Pipe {
+                fds.push((2, 2));
+                None
+            } else {
+                let io = Io::new(io).await?;
+                fds.push((io.0, 2));
+                Some(io)
+            }
+        } else {
+            None
+        };
+        if let Some(io) = manifest.io.as_ref().and_then(|io| io.stdin.as_ref()) {
+            if io == &IoInput::Pipe {
```

Comment:
it's probably only my lacking linux knowledge but this is hard to read without any comment

Reply:
In case of `IoOutput::Pipe` just add the fd from the rt (1 or 2) in the child process. This is done by add a (1, 1) or (2, 2) to the list of `preserved_fds`.

Otherwise create a `Io` that provides a fd that is mapped to 1 or 2 in the child.